spring-cloud-alibaba-starters/spring-cloud-starter-alibaba-nacos-discovery/src/main/java/com/alibaba/cloud/nacos/NacosServiceInstance.java

```diff
@@ -99,4 +100,27 @@ public void setMetadata(Map<String, String> metadata) {
 		this.metadata = metadata;
 	}
 
+	@Override
+	public boolean equals(Object o) {
+		if (this == o) {
+			return true;
+		}
+		if (o == null || getClass() != o.getClass()) {
+			return false;
+		}
+		NacosServiceInstance that = (NacosServiceInstance) o;
+
+		return Objects.equals(this.serviceId, that.serviceId)
+				&& Objects.equals(this.instanceId, that.instanceId)
+				&& Objects.equals(this.host, that.host)
+				&& this.port == that.port
+				&& this.secure == that.secure
+				&& Objects.equals(this.metadata, that.metadata);
+	}
+
+	@Override
+	public int hashCode() {
+		return (instanceId == null) ? 31 : (instanceId.hashCode() + 31);
```

Comment:
Why did you calculate hashCode by (instanceId.hashCode() + 31)? Whether just refer to the processing method of the InstanceInfo class?

Reply:
It's simple and effective
https://stackoverflow.com/questions/299304/why-does-javas-hashcode-in-string-use-31-as-a-multiplier